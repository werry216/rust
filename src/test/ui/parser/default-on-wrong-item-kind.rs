// Test parsing for `default` where it doesn't belong.
// Specifically, we are interested in kinds of items or items in certain contexts.

fn main() {}

#[cfg(FALSE)]
mod free_items {
    default extern crate foo; //~ ERROR item cannot be `default`
    default use foo; //~ ERROR item cannot be `default`
    default static foo: u8; //~ ERROR item cannot be `default`
    default const foo: u8; //~ ERROR item cannot be `default`
    default fn foo(); //~ ERROR item cannot be `default`
    default mod foo {} //~ ERROR item cannot be `default`
    default extern "C" {} //~ ERROR item cannot be `default`
    default type foo = u8; //~ ERROR item cannot be `default`
    default enum foo {} //~ ERROR item cannot be `default`
    default struct foo {} //~ ERROR item cannot be `default`
    default union foo {} //~ ERROR item cannot be `default`
    default trait foo {} //~ ERROR item cannot be `default`
    default trait foo = Ord; //~ ERROR item cannot be `default`
    default impl foo {}
    default!();
    default::foo::bar!();
    default macro foo {} //~ ERROR item cannot be `default`
    default macro_rules! foo {} //~ ERROR item cannot be `default`
}
