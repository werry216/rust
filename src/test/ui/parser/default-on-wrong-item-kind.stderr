error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:8:5
   |
LL |     default extern crate foo;
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:9:5
   |
LL |     default use foo;
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:10:5
   |
LL |     default static foo: u8;
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:11:5
   |
LL |     default const foo: u8;
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:12:5
   |
LL |     default fn foo();
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:13:5
   |
LL |     default mod foo {}
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:14:5
   |
LL |     default extern "C" {}
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:15:5
   |
LL |     default type foo = u8;
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:16:5
   |
LL |     default enum foo {}
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:17:5
   |
LL |     default struct foo {}
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:18:5
   |
LL |     default union foo {}
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:19:5
   |
LL |     default trait foo {}
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:20:5
   |
LL |     default trait foo = Ord;
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:24:5
   |
LL |     default macro foo {}
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: item cannot be `default`
  --> $DIR/default-on-wrong-item-kind.rs:25:5
   |
LL |     default macro_rules! foo {}
   |     ^^^^^^^ `default` because of this
   |
   = note: only associated `fn`, `const`, and `type` items can be `default`

error: aborting due to 15 previous errors

